packages/code-du-travail-frontend/src/common/SurveyModal.js

```diff
@@ -0,0 +1,469 @@
+import {
+  Button,
+  icons,
+  Input,
+  InputCheckbox,
+  InputRadio,
+  Modal,
+  Progress,
+  Text,
+  theme,
+} from "@socialgouv/cdtn-ui";
+import PropTypes from "prop-types";
+import React, { useCallback, useEffect, useReducer, useState } from "react";
+import styled from "styled-components";
+
+import { useLocalStorage } from "../lib/useLocalStorage";
+
+const PROMPT_DELAY = 15000; // 15 seconds
+
+const initialState = {
+  improvements: [],
+  page: 1,
+  status: "",
+  suggestion: "",
+  tailleEntreprise: "",
+};
+
+function reducer(state, action) {
```

Comment:
can we split that in its own file ? we could make a directory with, reducer + screen component

Reply:
Yep, let's split that